ProofWidgets/Component/RefreshComponent.lean

```diff
@@ -0,0 +1,213 @@
+/-
+Copyright (c) 2025 Jovan Gerbscheid. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jovan Gerbscheid
+-/
+import ProofWidgets.Data.Html
+import ProofWidgets.Util
+
+/-!
+## The `RefreshComponent` widget
+
+This file defines `RefreshComponent`, which allows you to have an HTML widget that updates
+incrementally as more results are computed by a Lean computation.
+
+For this interaction, we use an `IO.Ref` that the JavaScript reads from.
+It stores the HTML that should currently be on display, and a task returning the next HTML.
+To determine whether the widget is up to date, each computed HTML has an associated index.
+(So, the `n`-th HTML will have index `n`)
+
+When the widget (re)loads, it first loads the current HTML from the ref, and then
+repeatedly awaits further HTML result.
+-/
+
+namespace ProofWidgets
+open Lean Server Widget Jsx
+namespace RefreshComponent
+
+/-- The result that is sent to the `RefreshComponent` after each query. -/
+structure ResultProps where
```

Comment:
The convention in React is that the arguments to a component function are 'props', but other datatypes are not. This could be `VersionedHtml` or something like that.